src/software/ai/move_rule/move_rule_manager.cpp

```diff
@@ -0,0 +1,92 @@
+#include "software/ai/move_rule/move_rule_manager.h"
+
+#include "software/ai/hl/stp/tactic/all_tactics.h"
+
+
+std::set<MoveRule> MoveRuleManager::getMoveRules(const GameState &game_state,
+                                                 const Tactic &tactic)
+{
+    setCurrentMoveRulesFromGameState(game_state);
+    tactic.accept(*this);
+    return current_move_rules;
+}
+
+void MoveRuleManager::visit(const CherryPickTactic &tactic) {}
```

Comment:
How would we do whitelisting with this? Don't we need to add a "whitelist" that we can add areas to get filtered out after `setCurrentMoveRulesFromGameState` is called?

Reply:
`visit` will call erase on `current_move_rules` to "whitelist" things

Reply:
this has been replaced by remove_move_rule